clang/test/CIR/CodeGen/atomic.cpp

```diff
@@ -378,4 +378,96 @@ void inc_short(short* a, short b) {
 // LLVM: atomicrmw add ptr {{.*}}, i8 {{.*}} seq_cst, align 1
 void inc_byte(char* a, char b) {
   char c = __sync_fetch_and_add(a, b);
+}
+
+
+// CHECK-LABEL: @_Z12cmp_bool_int
+// CHECK: %[[PTR:.*]] = cir.load {{.*}} : !cir.ptr<!cir.ptr<!s32i>>, !cir.ptr<!s32i>
+// CHECK: %[[CMP:.*]] = cir.load {{.*}} : !cir.ptr<!s32i>, !s32i
+// CHECK: %[[UPD:.*]] = cir.load {{.*}} : !cir.ptr<!s32i>, !s32i
+// CHECK: %[[OLD:.*]], %[[RES:.*]] = cir.atomic.cmp_xchg(%[[PTR]] : !cir.ptr<!s32i>, %[[CMP]] : !s32i, %[[UPD]] : !s32i, success = seq_cst, failure = seq_cst) : (!s32i, !cir.bool)
+// CHECK: cir.store %[[RES]], {{.*}} : !cir.bool, !cir.ptr<!cir.bool>
+
+// LLVM-LABEL: @_Z12cmp_bool_int
+// LLVM: %[[PTR:.*]] = load ptr, ptr %4
+// LLVM: %[[CMP:.*]] = load i32, ptr %5
+// LLVM: %[[UPD:.*]] = load i32, ptr %6
```

Comment:
Can we do something better about `%4, %5, %6` here? Perhaps just cut the output after the comma?